modules/webgpu/src/adapter/resources/webgpu-buffer.ts

```diff
@@ -54,7 +54,7 @@ export class WebGPUBuffer extends Buffer {
     );
   }
 
-  override async readAsync(byteOffset: number = 0, byteLength: number = this.byteLength): Promise<ArrayBuffer> {
+  override async readAsync(byteOffset: number = 0, byteLength: number = this.byteLength): Promise<Uint8Array> {
```

Comment:
Changed to match WEBGLBuffer implementation (and would be my slight preference anyway).

We could also consider making `readAsync` take a target Uint8Array (optional? required?) and writing data into that, so that calling this in the frame loop doesn't necessarily allocate a new buffer each time.